libs/runBatch.js

```diff
@@ -42,37 +69,64 @@ const progress = (i, arr) => {
 
 const runBatch = async ({
   script,
+  domains,
   domainsFile,
-  limit,
-  poolSize,
-  dryRun,
-  fromDomain
+  limit = null,
+  poolSize = 10,
+  dryRun = true,
+  fromDomain = null,
+  verbose = true
 }) => {
   await config.loadConfig()
-  let domains = fs
-    .readFileSync(domainsFile)
-    .toString()
-    .split('\n')
-    .filter(x => x != '')
+  if (!domains && !domainsFile) {
+    throw new Error(
+      'runBatch: Invalid arguments. Need at least `domains` or `domainsFile`'
+    )
+  }
+  domains =
+    domains ||
+    fs
+      .readFileSync(domainsFile)
+      .toString()
+      .split('\n')
+      .filter(x => x != '')
 
   if (fromDomain) {
     const i = domains.findIndex(domain => domain === fromDomain)
     domains = domains.slice(i)
   }
 
+  // An object that will be passed to functions run on each cozy
+  // Can be used to store global stats
+  const stats = {}
+
+  const globalCtx = {
+    stats,
+    dryRun,
+    verbose
+  }
+
   const start = new Date()
   const pool = new PromisePool(
     runScriptPool(
       script,
       limit ? domains.slice(0, limit) : domains,
       progress,
-      dryRun
+      globalCtx
     ),
     poolSize
   )
+  const results = []
+  pool.addEventListener('fulfilled', function(event) {
+    results.push(event.data.result)
+  })
```

Comment:
It could be useful to remove this with an option at some point since now the script runs in O(n) in memory.